src/main/java/service/impl/ShopServiceImpl.java

```diff
@@ -0,0 +1,42 @@
+package service.impl;
+
+import database.Storage;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import model.FruitTransaction;
+import service.ShopService;
+import strategy.OperationStrategy;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> fruitTransactions) {
+        Map<String, Integer> toStorage = fruitTransactions.stream()
+                .collect(Collectors.toMap(
+                        FruitTransaction::getFruit,
+                        this::getQuantityOfTransaction,
+                        Integer::sum
+                ));
+        checkBalance(toStorage);
+        Storage.updateDataBase(toStorage);
```

Comment:
Avoid hardcoding method calls in your services. Instead of directly calling `Storage.updateDataBase(toStorage);` here, you can pass the `Storage` object as a parameter in the constructor of this service. This will make your service more flexible and easier to test.